README.md

```diff
@@ -93,10 +83,21 @@ When you're done testing the changes locally, here is how you release a new vers
     1. Using GitHub: Go to the [releases page](/releases) and click "Draft a new release".
     1. Using Git:
 
-    ```
+    ```sh
     git tag -a v0.0.2 -m "tag message"
     git push --follow-tags
     ```
+
 1. Now you can use the new Git tag (e.g. `v0.0.2`) in the `ref` attribute of the `source` URL in `terragrunt.hcl`.
 1. Run `terragrunt plan`.
-1. If the plan looks good, run `terragrunt apply`.   
+1. If the plan looks good, run `terragrunt apply`.
+
+## Monorepo vs. polyrepo
+
+This example shows live modules in a "monorepo" - several modules in a single repository. There are [benefits and drawbacks to using a monorepo](https://github.com/joelparkerhenderson/monorepo_vs_polyrepo) vs. using a "polyrepo" - one module per repository. Which you choose depends on your tooling, how you build/test Terraform modules, and so on.
+
+For example, if you have a CI pipeline that automates testing on Terraform modules, a monorepo will either need to test all the modules in the repo for any change; or have tooling that supports only testing things that changed. A polyrepo scenario doesn't require such tooling.
```

Comment:
Perhaps we should list our own set of pros/cons. Example:

**Advantages of a monorepo for Terraform modules**

* Easier to make global changes across the entire codebase. E.g., Applying a critical security fix or upgrading everything to a new version of Terraform.
* Easier to search across the entire codebase. Reduce the odds of having 300 copies of a `StringUtil` class.
* Continuous integration: all your code is tested and versioned together. Less chance of some developer using an ancient version of a dependency who can never update. 
* One repo and CI builds to manage.

**Disadvantages of a monorepo for Terraform modules**

* Harder to make local changes, as while you're modifying module `foo`, you also have to think through whether this will affect module `bar`. 
* The simple approach is to run all tests after every commit, but as the monorepo grows, this gets slower and slower (and more brittle). 
* To only run a subset of the tests, you need a way to tell which modules were affected by which commits. Unfortunately, Terraform has no first-class dependency management system, so there's no way to know that a code change in a file in module `foo` won't affect module `bar`. So you have to build custom tooling that figures this out based on heuristics (brittle) or try to integrate Terraform with dependency management / monorepo tooling like bazel (lots of work).
* Doesn't work with the Terraform Private Registry (part of Terraform Enterprise and Terraform Cloud), which requires one module per repo.
* Terraform doesn't support feature toggles, which are often critical for making large scale changes in a monorepo.

**Advantages of one-repo-per-module**

* Easier to make local changes, as you mostly only have to think about the one module/repo you're changing.
* Works with the Terraform Private Registry (part of Terraform Enterprise and Terraform Cloud), assuming you put the Terraform code in the root of the repo and follow their naming conventions.
* When you run tests, it's just tests for this one module, so no extra tooling is necessary to keep tests fast.

**Disdvantages of one-repo-per-module**

* Harder to make global changes, as you have to check out tons of repos, make many separate commits & PRs, convince each team to merge the changes, release tons of new versions, and do it all in a very specific order based on the dependency graph. This can take _months_ in a large organization, which is hugely problematic when dealing with security vulnerabilities. 
* Harder to search across the codebase, as you again have to check out tons of repos. Results in a lot of duplicated code/effort.
* No continuous integration. You might make a change in your module, and the teams that depend on you might not consume that change for months... And when they do, they find some nasty backwards incompatibility, which is now very hard to fix after so much time has changed.
* Many repos and CI builds to manage.
* Dependency graph problems. E.g., Look up diamond dependency problems. Note that this mostly does not affect Terraform, as it can run different versions of the same dependency at the same time, so I'm just listing it here for completeness.

Reply:
> Reduce the odds of having 300 copies of a StringUtil class.

This doesn't sound like a problem we'd have in Terraform. Maybe "Reduce the odds of having 300 ways to deploy the same resource?" Can you use classes in Terraform modules?

Reply:
Updated the pros/cons with inline descriptions. I started with the ones you listed and added minor verbiage updates. Also added a pro/con about versioning/releases in a monorepo vs polyrepo.

Reply:
> This doesn't sound like a problem we'd have in Terraform. Maybe "Reduce the odds of having 300 ways to deploy the same resource?" Can you use classes in Terraform modules?

Haha, yea, `StringUtil` is just one that everyone has likely seen in other languages 😁 

The equivalent in Terraform is indeed having 300 ways to deploy the same resource or type of infrastructure. E.g., We've seen large companies who had a dozen nearly-identical-but-not-quite RDS modules.